Gems/Atom/RHI/Code/Include/Atom/RHI/RHISystem.h

```diff
@@ -64,6 +64,7 @@ namespace AZ::RHI
         //////////////////////////////////////////////////////////////////////////
         // RHISystemInterface Overrides
         RHI::Device* GetDevice(int deviceIndex = MultiDevice::DefaultDeviceIndex) override;
+        [[nodiscard]] AZStd::optional<int> AddVirtualDevice(int deviceIndexToVirtualize = MultiDevice::DefaultDeviceIndex) override;
```

Comment:
Please add a comment explaining the purpose of the function. We are trying to follow this convention for all the new code going in.